datadog/fwprovider/resource_datadog_integration_gcp.go

```diff
@@ -0,0 +1,357 @@
+	state.CspmResourceCollectionEnabled = types.BoolValue(resp.GetIsCspmEnabled())
+	state.ResourceCollectionEnabled = types.BoolValue(resp.GetResourceCollectionEnabled())
+	state.IsSecurityCommandCenterEnabled = types.BoolValue(resp.GetIsSecurityCommandCenterEnabled())
+
+	// Non-computed values
+	if clientId, ok := resp.GetClientIdOk(); ok {
+		state.ClientId = types.StringValue(*clientId)
+	}
+	if privateKey, ok := resp.GetPrivateKeyOk(); ok {
+		state.PrivateKey = types.StringValue(*privateKey)
+	}
+	if privateKeyId, ok := resp.GetPrivateKeyIdOk(); ok {
+		state.PrivateKeyId = types.StringValue(*privateKeyId)
+	}
+	state.ID = projectId
+}
+
+func (r *integrationGcpResource) getGCPIntegration(state integrationGcpModel) (*datadogV1.GCPAccount, diag.Diagnostics) {
+	diags := diag.Diagnostics{}
+	resp, _, err := r.Api.ListGCPIntegration(r.Auth)
+	if err != nil {
+		diags.Append(utils.FrameworkErrorDiag(err, "error listing GCP integration"))
+		return nil, diags
+	}
+
+	var account *datadogV1.GCPAccount
+	for _, integration := range resp {
+		if integration.GetProjectId() == state.ProjectID.ValueString() && integration.GetClientEmail() == state.ClientEmail.ValueString() {
+			if err := utils.CheckForUnparsed(integration); err != nil {
+				diags.AddError("response contains unparsedObject", err.Error())
+				return nil, diags
+			}
+			account = &integration
+			break
+		}
+	}
+
+	return account, diags
+}
+
+func (r *integrationGcpResource) buildIntegrationGcpRequestBodyBase(state integrationGcpModel) (*datadogV1.GCPAccount, diag.Diagnostics) {
+	diags := diag.Diagnostics{}
+	body := datadogV1.NewGCPAccountWithDefaults()
+	body.SetProjectId(state.ProjectID.ValueString())
+	body.SetClientEmail(state.ClientEmail.ValueString())
+
+	return body, diags
+}
+
+func (r *integrationGcpResource) addDefaultsToBody(body *datadogV1.GCPAccount, state integrationGcpModel) {
+	body.SetType(defaultType)
+	body.SetAuthUri(defaultAuthURI)
+	body.SetAuthProviderX509CertUrl(defaultAuthProviderX509CertURL)
+	body.SetClientX509CertUrl(defaultClientX509CertURLPrefix + state.ClientEmail.ValueString())
+	body.SetTokenUri(defaultTokenURI)
+}
+
+func (r *integrationGcpResource) addRequiredFieldsToBody(body *datadogV1.GCPAccount, state integrationGcpModel) {
+	body.SetClientId(state.ClientId.ValueString())
+	body.SetPrivateKey(state.PrivateKey.ValueString())
+	body.SetPrivateKeyId(state.PrivateKeyId.ValueString())
+}
+
+func (r *integrationGcpResource) addOptionalFieldsToBody(body *datadogV1.GCPAccount, state integrationGcpModel) {
+	body.SetAutomute(state.Automute.ValueBool())
+	body.SetIsCspmEnabled(state.CspmResourceCollectionEnabled.ValueBool())
+	body.SetIsSecurityCommandCenterEnabled(state.IsSecurityCommandCenterEnabled.ValueBool())
+	body.SetHostFilters(state.HostFilters.ValueString())
+	if !state.ResourceCollectionEnabled.IsUnknown() {
+		body.SetResourceCollectionEnabled(state.ResourceCollectionEnabled.ValueBool())
+	}
```

Comment:
Any reason this is special case? (i.e. I don't see `IsUnknown()` used anywhere else but `ResourceCollectionEnabled`)

Reply:
Mainly for backwards compatibility. The API will return an error if the CSPM flag is on but the Resource Collection Flag is explicitly set to False. In this case, we don't want to set any value for the Resource Collection flag unless a value was explicitly set by the user.